### PR TITLE
Remaining prow jobs migration

### PR DESCRIPTION
**Pull request by kolodziejczak**
**Description**

Changes proposed in this pull request:

- Remaining migration requiring jobs moved from prow

### CODE REVIEW

File: .github/workflows/pull-integration-release.yaml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+          fetch-depth: 0
+      - uses: ./.github/actions/upgrade-test
+        with:
+          test_upgrade_img: "europe-docker.pkg.dev/kyma-project/dev/api-gateway-manager:PR-${{github.event.number}}"
+          client_id: ${{ secrets.CLIENT_ID }}
+          client_secret: ${{ secrets.CLIENT_SECRET }}
+          oidc_issuer_url: ${{ secrets.OIDC_ISSUER_URL }}
+          target_branch: ${{ github.base_ref }}
+
+  custom-domain-gcp:
```
Comment (barchw) [inline .github/workflows/pull-integration-release.yaml:87]: Custom domain is not supposed to run on PRs

File: .github/workflows/pull-integration.yaml [hunk 2]
```diff
@@ -18,7 +18,6 @@ on:
       - ".github/**" #Ignoring since build job isn't triggered on the workflow directory changes
     branches:
       - main
-      - 'release-**'
```
Comment (barchw) [inline .github/workflows/pull-integration.yaml:21]: Why is this removed?

File: tests/integration/scripts/custom-domain-gardener-gcp-gh.sh [hunk 3]
```diff
@@ -16,6 +16,6 @@ export GARDENER_REGION="europe-west3"
 export GARDENER_ZONES="europe-west3-c,europe-west3-b,europe-west3-a"
 export GARDENER_PROVIDER_SECRET_NAME="goat"
 export GARDENER_PROJECT_NAME="goatz"
-export GARDENER_CLUSTER_VERSION="1.26.9"
+export GARDENER_CLUSTER_VERSION="1.26.11"
```
Comment (barchw) [inline tests/integration/scripts/custom-domain-gardener-gcp-gh.sh:19]: Higher minor should also work

File: .github/workflows/main-integration.yaml [hunk 4]
```diff
@@ -90,6 +90,25 @@ jobs:
...
+      - uses: ./.github/actions/integration-test-gardener
+        with:
+          manager_image: "europe-docker.pkg.dev/kyma-project/prod/api-gateway-manager:${{ needs.wait-for-image-build.outputs.sha }}"
+          gardener_secret: ${{ secrets.GARDENER_TOKEN }}
+          script: ./tests/integration/scripts/custom-domain-gardener-aws.sh
+          client_id: ${{ secrets.CLIENT_ID }}
+          client_secret: ${{ secrets.CLIENT_SECRET }}
+          oidc_issuer_url: ${{ secrets.OIDC_ISSUER_URL }}
+          dns_secret_json: ${{ secrets.DNS_SECRET_JSON }}
+
```
Comment (barchw) [inline .github/workflows/main-integration.yaml:111]: Why no GCP?

**Review by barchw — CHANGES_REQUESTED** [review]

File: .github/workflows/pull-integration.yaml (hunk 2 above)
Comment (kolodziejczak) [reply .github/workflows/pull-integration.yaml:21]: PR provides another workflow for pull requests to the release branch

File: .github/workflows/pull-integration-release.yaml (hunk 1 above)
Comment (kolodziejczak) [reply .github/workflows/pull-integration-release.yaml:87]: It is supposed to run only on PRs to the release branch. This is also why PR provides separate workflow for PRs pointing to the release branches

Comment (barchw) [reply .github/workflows/pull-integration-release.yaml:87]: How do we want to support that without `pullRequestTarget`?
Does that mean that we require to do PRs to release branches only from same-repo?

Comment (barchw) [reply .github/workflows/pull-integration-release.yaml:87]: OK

**Review by barchw — APPROVED** [review]
